### ISSUE TITLE
CA2101: False positive when all strings are marshaled as UTF-8.

### ISSUE THREAD

**Issue by nguerrera** [dotnet/roslyn-analyzers#7502]
### Analyzer

**Diagnostic ID**: [CA2101](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2101): `Specify marshaling for P/Invoke string arguments.`

### Analyzer source

**SDK**: Built-in CA analyzers in .NET 9.0.200 Preview SDK

**Version**: 9.0.200-preview.0.24604.10

### Steps To Reproduce

* Run the analyzer against this:
```C#
[DllImport("example")]
static extern DoSomething([MarshalAs(UnmanagedType.LPUTF8Str)] string value);
```
### Expected behavior

CA2101 does not warn.

### Actual behavior

CA2101 warns.

## Additional context

The rule doc says:

> When you convert from Unicode to ANSI, it is possible that not all Unicode characters can be represented

This does not apply here there is no conversion to ANSI when using UTF8. I could set BestFitMapping=false as a workaround, but that confuses matters IMHO since there wouldn't be any BestFitMapping anyway if all strings are marshaled as UTF-8.

I believe the rule should allow `LPUTF8Str` wherever it allows `LPWStr`.
